framework/src/org/apache/cordova/SystemBarPlugin.java

```diff
@@ -0,0 +1,289 @@
+    private int overrideStatusBarBackgroundColor = INVALID_COLOR;
+
+
+    @Override
+    protected void pluginInitialize() {
+        context = cordova.getContext();
+        resources = context.getResources();
+    }
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        cordova.getActivity().runOnUiThread(this::updateSystemBars);
+    }
+
+    @Override
+    public void onResume(boolean multitasking) {
+        super.onResume(multitasking);
+        cordova.getActivity().runOnUiThread(this::updateSystemBars);
+    }
+
+    @Override
+    public Object onMessage(String id, Object data) {
+        if (id.equals("updateSystemBars")) {
+            cordova.getActivity().runOnUiThread(this::updateSystemBars);
+        }
+        return null;
+    }
+
+    @Override
+    public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM
+                && preferences.getBoolean("AndroidEdgeToEdge", false)
+        ) {
+            // Disable JS API in E2E mode (SDK >= 35)
+            return false;
+        }
+
+        if ("setStatusBarVisible".equals(action)) {
+            boolean visible = args.getBoolean(0);
+            cordova.getActivity().runOnUiThread(() -> setStatusBarVisible(visible));
+        } else if ("setStatusBarBackgroundColor".equals(action)) {
+            String bgColor = args.getString(0);
+            cordova.getActivity().runOnUiThread(() -> setStatusBarBackgroundColor(bgColor));
+        } else {
+            return false;
+        }
+
+        callbackContext.success();
+        return true;
+    }
+
+    private void setStatusBarVisible(final boolean visible) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.VANILLA_ICE_CREAM) {
+            View statusBar = getStatusBarView(webView);
+            if (statusBar != null) {
+                statusBar.setVisibility(visible ? View.VISIBLE : View.GONE);
+
+                FrameLayout rootLayout = getRootLayout(webView);
+                if (rootLayout != null) {
+                    ViewCompat.requestApplyInsets(rootLayout);
+                }
+            }
+        } else {
+            Window window = cordova.getActivity().getWindow();
+            int uiOptions = window.getDecorView().getSystemUiVisibility();
+            int flags = View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_FULLSCREEN;
+            if (visible) {
+                uiOptions &= ~flags;
+                window.clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+            } else {
+                uiOptions |= flags;
+                window.addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+            }
+            window.getDecorView().setSystemUiVisibility(uiOptions);
+        }
+    }
+
+    private void setStatusBarBackgroundColor(final String colorPref) {
```

Comment:
For iOS, I passed the colour values as 3 integers, which allowed a bit more flexibility in how it was specified on the JS side (i.e., to allow setting it to `rgb(123 234 124)` or `maroon` and using CSS.getComputedStyle to parse it and guarantee we'd get 3 RGB components)